### PR TITLE
Issue #13345: Enable test case for parameternameChecktest

### PR DESCRIPTION
**Pull request by Rohanraj123**
part of #13345

### CODE REVIEW

**Comment by Rohanraj123** [comment]
Don't worry I know where's the problem. I will fix this soon.

**Comment by romani** [comment]
Before push to GitHub, please run `mvn clean verify` on your local.

Without this you will waste a lot of time.

**Comment by Rohanraj123** [comment]
@romani Can we put `HacktoberFest `Label on this . Please

**Comment by romani** [comment]
@Rohanraj123 , label was added, please recheck that it is good.

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/parametername/Example5.java [hunk 1]
```diff
@@ -5,25 +5,25 @@
       <property name="format" value="^[a-z]([a-z0-9][a-zA-Z0-9]*)?$"/>
       <property name="accessModifiers" value="protected, package, private"/>
       <message key="name.invalidPattern"
-        value="Parameter name ''{0}'' must match pattern ''{1}''"/>
+        value="Name ''{0}'' must match pattern ''{1}''."/>
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/parametername/Example5.java:8]: please revert this update.
It was done on purpose to show that user can override message
https://github.com/checkstyle/checkstyle/blob/56af76110fd2b6acee4e47db63b7bd4511dc6c15/src/main/resources/com/puppycrawl/tools/checkstyle/checks/naming/messages.properties#L4
https://checkstyle.org/checks/naming/parametername.html#Violation_Messages

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/parametername/Example5.java [hunk 2]
```diff
@@ -5,25 +5,25 @@
...
-class MyClass {
-  void method1(int v1) {} // OK
-  protected method2(int V2) {} // violation, Parameter name 'V2'
-                               // must match pattern '^[a-z]([a-z0-9][a-zA-Z0-9]*)?$'
-  private method3(int a) {} // OK
-  public method4(int b) {} // violation, Parameter name 'b'
-                           // must match pattern '^[a-z][a-z0-9][a-zA-Z0-9]*$'
+class Example5 {
+  void method1(int v1) {}
+  protected void method2(int V) {} // violation
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/parametername/Example5.java:25]: in this specific case when message is defined by user in confg, please keep trailing comment and make it like: 
`// violation "Parameter name 'V2' must match pattern"`

**Review by romani — CHANGES_REQUESTED** [review]
items:

**Comment by Rohanraj123** [comment]
`  protected void method2(int V) {} // violation "Parameter name 'V' must match pattern '^[a-z]([a-z0-9][a-zA-Z0-9]*)?$'"` when i write this line in Example5.java file and revert the updates as you suggested

```
  expected to match: 25:(?:\d+:)?\s.*Parameter name 'V' must match '^[a-z]\([a-z0-9][a-zA-Z0-9]*\)?$'.*
but was          : 25:30: Parameter name 'V' must match pattern '^[a-z]([a-z0-9][a-zA-Z0-9]*)?$'
```

This is the issue appears while `mvn clean verify`
thats why I updated the message in `parametername.xml`
Can you suggest me some solution? Please

**Comment by romani** [comment]
Please try as I suggested originally `// violation "Parameter name 'V' must match pattern"`

**Comment by Rohanraj123** [comment]
> Please try as I suggested originally `// violation "Parameter name 'V' must match pattern"`
> 
> Pattern is skipped intentionally

As per the suggested changes 

`  public void method4(int b) {} // violation "Parameter name 'b' must match pattern"`

Output : 
```
--
expected      : [25:30: Name 'V' must match pattern '^[a-z]([a-z0-9][a-zA-Z0-9]*)?$'., 27:27: Name 'b' must match pattern '^[a-z][a-z0-9][a-zA-Z0-9]*$'.]
but was       : [25:30: Parameter name 'V' must match pattern '^[a-z]([a-z0-9][a-zA-Z0-9]*)?$', 27:27: Parameter name 'b' must match pattern '^[a-z][a-z0-9][a-zA-Z0-9]*$']

```

**Comment by Rohanraj123** [comment]
Either we need to to customisation in` parametername.xml` file in violation message or we need to make custom` expected `i.e 
```

@Test
public void testExample5() throws Exception {
    final String[] expected = {
        "25:30: " + getCheckMessage("Name 'V' must match pattern '^[a-z]([a-z0-9][a-zA-Z0-9]*)?$'."),
        "27:27: " + getCheckMessage("Name 'b' must match pattern '^[a-z][a-z0-9][a-zA-Z0-9]*$'."),
    };

    verifyWithInlineConfigParser(getPath("Example5.java"), expected);
}
```

**Comment by romani** [comment]
>or we need to make customexpected

Yes.

**Comment by Rohanraj123** [comment]
```
@Test
    public void testExample5() throws Exception {
        final String[] expected = {
            "25:30: " + "Parameter name 'V' must match pattern '^[a-z]([a-z0-9][a-zA-Z0-9]*)?$'",
            "27:27: " + "Parameter name 'b' must match pattern '^[a-z][a-z0-9][a-zA-Z0-9]*$'",
        };

        verifyWithInlineConfigParser(getPath("Example5.java"), expected);
    }
```
@romani Sir, is it okay?

**Comment by Rohanraj123** [comment]
> ```
> @Test
>     public void testExample5() throws Exception {
>         final String[] expected = {
>             "25:30: " + "Parameter name 'V' must match pattern '^[a-z]([a-z0-9][a-zA-Z0-9]*)?$'",
>             "27:27: " + "Parameter name 'b' must match pattern '^[a-z][a-z0-9][a-zA-Z0-9]*$'",
>         };
> 
>         verifyWithInlineConfigParser(getPath("Example5.java"), expected);
>     }
> ```
> 
> @romani Sir, is it okay?

After this `mvn clean verify` check is successful

**Review by romani — APPROVED** [review]
Ok to merge

File: src/xdocs/checks/naming/parametername.xml [hunk 3]
```diff
@@ -179,13 +179,11 @@ class MyClass {
...
-class MyClass {
-  void method1(int v1) {} // OK
-  protected method2(int V2) {} // violation, Parameter name 'V2'
-                               // must match pattern '^[a-z]([a-z0-9][a-zA-Z0-9]*)?$'
-  private method3(int a) {} // OK
-  public method4(int b) {} // violation, Parameter name 'b'
-                           // must match pattern '^[a-z][a-z0-9][a-zA-Z0-9]*$'
+class Example5 {
+  void method1(int v1) {}
+  private void method2(int V) {} // violation &quot;Parameter name 'V' must match pattern&quot;
```
Comment (rnveach) [inline src/xdocs/checks/naming/parametername.xml:184]: Why did we lose protected?
Also why did we lose the 2 in the parameter name?

Comment (Rohanraj123) [reply src/xdocs/checks/naming/parametername.xml:184]: just to maintain the required length of line as fixed in the config. of checkstyle. that was `85` or something. But the lenght was exceeding the maximum extent and was `87` or something. `private` and `protected` following the same pattern thats why I thought to make it private. Also in the `ParameterNameCheckTestExample.java` file line length was exceeding maxium extent of `100` words thats why I changed `V2` to `V`

Comment (rnveach) [reply src/xdocs/checks/naming/parametername.xml:184]: The other examples in this PR removed the violation messages except for this one file.

@romani Are we keeping the violation messages or not?

Comment (romani) [reply src/xdocs/checks/naming/parametername.xml:184]: We keep only when it matters.
In this check only when we override message.
In other cases it is too obvious.

Comment (romani) [reply src/xdocs/checks/naming/parametername.xml:184]: >Why did we lose protected?
Also why did we lose the 2 in the parameter name?

This is not expected. We need to restore.
We can always shorten something else in method declaration or...

Please share a problem and we will suggest how to handle it.

Comment (Rohanraj123) [reply src/xdocs/checks/naming/parametername.xml:184]: Can we change the methods name here to maintain the line size issue?

Comment (Rohanraj123) [reply src/xdocs/checks/naming/parametername.xml:184]:
```
class Example5 {
  void fun1(int v1) {}
  protected void fun2(int V2) {} // violation "Parameter name 'V' must match pattern"
  private void fun3(int a) {}
  public void fun4(int b) {} // violation "Parameter name 'b' must match pattern"
```

Like this?

Comment (romani) [reply src/xdocs/checks/naming/parametername.xml:184]: good for me, please put this in code to see in diff more clearly that nothing is lost

Comment (Rohanraj123) [reply src/xdocs/checks/naming/parametername.xml:184]: Okay

Comment (Rohanraj123) [reply src/xdocs/checks/naming/parametername.xml]: Done with the changes. Review it once please

**Review by rnveach — APPROVED** [review]

**Pull request merged**